### PR TITLE
[Order] Automatically cancel unpaid orders

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets |
| License         | MIT

This PR is based on https://github.com/Sylius/Sylius/pull/6291.

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
@Zales0123 Please rebase. :)

**Comment by Zales0123** [comment]
@pjedrzejewski Done, waiting for review 🌮

**Comment by pjedrzejewski** [comment]
Thanks Mateusz, nice work! :) 👍

File: features/cart/shopping_cart/deleting_expired_carts_automatically.feature [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+Feature: Deleting expired carts automatically
+    In order to get rid of started but not finished orders
+    As an Administrator
+    I want to have expired carts automatically deleted
+
+    Background:
+        Given the store operates on a single channel in "United States"
+
+    @domain
+    Scenario: Having cart deleted after 3 hours of idleness
```
Comment (Zales0123) [inline features/cart/shopping_cart/deleting_expired_carts_automatically.feature]: :( What a horrible world would it be -> I will fix this in a separate PR together with fixes from https://github.com/Sylius/Sylius/pull/6291

File: features/order/managing_orders/cancelling_unpaid_orders_automatically.feature [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store has a product "PHP T-Shirt"
+        And the store ships everywhere for free
+        And the store allows paying with "Paypal Express Checkout"
+
+    @domain
+    Scenario: Having order cancelled after 10 days of being unpaid
+        And there is a customer "john.doe@gmail.com" that placed an order "#00000022"
```
Comment (lchrusciel) [inline features/order/managing_orders/cancelling_unpaid_orders_automatically.feature]: Given

File: features/order/managing_orders/cancelling_unpaid_orders_automatically.feature [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        And the customer bought a single "PHP T-Shirt"
+        And the customer chose "Free" shipping method to "United States" with "Paypal Express Checkout" payment
+        And this order has not been paid for 10 days
+        Then this order should be automatically cancelled
+
+    @domain
+    Scenario: Having unpaid order not cancelled if expiration time has not been reached
+        And there is a customer "john.doe@gmail.com" that placed an order "#00000022"
```
Comment (lchrusciel) [inline features/order/managing_orders/cancelling_unpaid_orders_automatically.feature]: Given

File: features/order/managing_orders/cancelling_unpaid_orders_automatically.feature [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        And the customer bought a single "PHP T-Shirt"
+        And the customer chose "Free" shipping method to "United States" with "Paypal Express Checkout" payment
+        And this order has not been paid for 2 days
```
Comment (lchrusciel) [inline features/order/managing_orders/cancelling_unpaid_orders_automatically.feature]: Maybe `When` would be more meaningful

File: src/Sylius/Bundle/OrderBundle/Command/RemoveExpiredCartsCommand.php [hunk 5]
```diff
@@ -36,6 +36,11 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        $expirationTime = $this->getContainer()->getParameter('sylius_order.cart_expiration_period');
+        $output->writeln(
+            sprintf('Command will remove carts that have been idle for <info>%s</info>.', $expirationTime)
```
Comment (lchrusciel) [inline src/Sylius/Bundle/OrderBundle/Command/RemoveExpiredCartsCommand.php]: `Command will remove carts that have not been paid for` as it is a main purpose of this command

**Comment by lchrusciel** [comment]
I haven't seen [this comment](https://github.com/Sylius/Sylius/pull/6321#discussion_r82265386). Nevermind :)

File: src/Sylius/Bundle/OrderBundle/Command/RemoveExpiredCartsCommand.php (hunk 5 above)
Comment (Zales0123) [inline src/Sylius/Bundle/OrderBundle/Command/RemoveExpiredCartsCommand.php]: Ok, in fact nope :D This command removes expired cart, that cannot even be paid (orders can be paid). Which means "idle" is a good word for me ;)

**Pull request merged**